backend/utils/db_utils.py

```
@@ -0,0 +1,61 @@
+                db.add(client)  # Add new client to the session
+                db.commit()  # Commit to save the new client
+
+            # Ensure the object is properly persisted
+            db.refresh(client)
+            return client
+
+    except Exception as e:
+        # Rollback in case of error
+        db.rollback()
+        print(f"Error during client add or update: {str(e)}")
+        raise Exception(f"Error during client add or update: {str(e)}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)